### ISSUE TITLE
Add unit tests

### ISSUE THREAD

**Issue by aawgit** [aawgit/onlineStore#15]
Adding unit tests will be a good value addition to the project.

**Comment by amatyas001** [comment]
Hello! I have started adding the test. I can see there is some not-too-good solutions in code and missing part. I am working on these and push them to review when it is in deployable state.

**Comment by aawgit** [comment]
Hello,
Thanks. I'm doing some refactoring to facilitate unit tests. Please wait
till I merge those to the master branch.
Thanks again.

**Comment by aawgit** [comment]
@amatyas001 Refactoring is done.

**Comment by uaalumni02** [comment]
Has the testing been complete?

**Comment by aawgit** [comment]
No, tests can be added.

**Comment by discoveranirban** [comment]
Hey! can Test still be added? I'm looking forward to contribute for the first time in an open source project. So is there a scope ?

**Comment by aawgit** [comment]
Yes, it's possible to add tests. I think back end code is a good place to start. I suggest using Jest, but feel free to bring up another framework if you prefer.

**Comment by discoveranirban** [comment]
Thanks for letting me know! I'll start if off and update you with my progress!

**Comment by discoveranirban** [comment]
is there anyway to contact you as i had some doubts because it's my first time.

**Comment by aawgit** [comment]
No worries. Feel free to contact me via email - akalankaweerasooriya@gmail.com, or LinkedIn: https://www.linkedin.com/in/akalanka-weerasooriya-7812b3171/

**Comment by ampsteric** [comment]
is this issue completed?

**Comment by phoneix007** [comment]
Hello, is this issue resolved?bcoz I am looking forward  contribute , Please let me know.

**Comment by phoneix007** [comment]
Could you give me some guidelines to start.

**Comment by aawgit** [comment]
> Could you give me some guidelines to start.

Sure, let me add a few tests, then it will be easier for you to start. It might take a couple of days. I'll keep you posted.

**Comment by aawgit** [comment]
> Hello, is this issue resolved? bcoz I am looking forward to contribute in this, Please let me know.

There is still room to add tests.

**Comment by aawgit** [comment]
> is this issue resolved?

Not yet.

**Comment by aawgit** [comment]
> Could you give me some guidelines to start.

@phoneix007 I have added one test under server/test/controller. Please feel free to contact me if you need any clarification.

**Comment by ranaDherya** [comment]
hey @aawgit, is this issue still up to take?

**Comment by aawgit** [comment]
> hey @aawgit, is this issue still up to take?

Yeah, why not.

**Comment by UpperLEFTY** [comment]
@aawgit Are you still in need of more unit tests? I would love to contribute!

**Comment by aawgit** [comment]
> @aawgit Are you still in need of more unit tests? I would love to contribute!

Yes

**Comment by 15Arindam** [comment]
Hey, I'd like to try this, please let me know if I can contribute

**Comment by Abhigpt211** [comment]
I'm interested to contribute on this project. How can I start?

**Comment by cjordan223** [comment]
Hi, can I contribute to this project?

**Comment by rudrikasharma15** [comment]
Hi @aawgit, I’d love to work on this issue and contribute unit tests to the backend! Could you please assign it to me? 🙌

**Comment by NEKKALAPUSIVASATYASAILIKHITHA** [comment]
Hello @aawgit i would like to contribute
